### ISSUE TITLE
Grid view: Change columns with pinch zoom (like google fotos)

### ISSUE THREAD

**Issue by tobiasKaminsky** [owncloud/android#1402]
currently it is hardcoded to three colums if I remember correctly.
What about a slider which changes this directly between 3 and 10 (or something higher)?

@jancborchardt Where can we put this? It should be directly in the grid view? Behind the three dots?

**Comment by jancborchardt** [comment]
I would rather not have an option for it as it’s something the app should take care of and be properly designed. We currently do a nice job of adjusting to portrait and landscape (3 columns in portrait on my device, and 5 in landscape).

**Comment by tobiasKaminsky** [comment]
Hm...What if you want to browse your images really quick? Then it would be great that one can adjust the column number...

**Comment by cornzy** [comment]
@tobiasKaminsky I would prefer just pinch zoom on the grid (like you have it in the gallery). No need for sliders and most intuitive. But I agree with @jancborchardt that it needs to be properly designed and it is not that easy.

**Comment by tobiasKaminsky** [comment]
@cornzy pinch zoom? In which gallery? I have never seen this.
But this would be the best way, I think.

**Comment by cornzy** [comment]
@tobiasKaminsky never seen or never tried ;) I had it in Sony's gallery app, now with my galaxy s6 in Samsung Gallery and even in Google Foto.

**Comment by tobiasKaminsky** [comment]
Google foto is doing a great job.
Thank you for pointing this out.
@jancborchardt I think this is the best way (if we want to implement it)

**Comment by jancborchardt** [comment]
Hm, sounds ok. We definitely need to set sensible max and min values as well as zoom steps. What does Google Photo do?

**Comment by AndyScherzinger** [comment]
Google Photos zooms time and size:
when you zoom out the images get smaller and it changes the grouping. Google photos uses sticky header where the smallest grouping is day and the larges is year. The groups gets more fine grained the more you scroll down (move back in time) and it also gets more coarse grained the more you zoom out. Very tricky but smart way (for photos)

**Comment by jancborchardt** [comment]
Ah ok, same as iPhoto on iOS then, grouping by event/location/time.

**Comment by AndyScherzinger** [comment]
Exactly. We do not have any grouping at all at the moment and I guess if we would like to introduce this we could only do time since anything else would be something that needs a lot of processing and if so should be something done on the server side (if ever). The other thing would be that we would have to add Headers to the grid view so the UI for images/videos would differ from any other.
So for starters we should probably leave out any grouping logic.

**Comment by jancborchardt** [comment]
Yeah. But in any case I don't see this as a high priority because the default is pretty good atm.

**Comment by tobiasKaminsky** [comment]
My idea was to "just" adjust the size to see more images at once. Nothing more, nothing less.
@jancborchardt Of course it is not high priority, but something I think is nice for me, so I am more into developing this than something else right now ;)

**Comment by jancborchardt** [comment]
Ok, then just to answer any spec questions in case you want to dive in :)
- Since we currently do 3 pictures in the width of a normal device (at least on my Moto G that’s the case), the second view should likely be 5 pictures in that width. That will effectively display about double the amount of pictures, or even more depending on the height.
- The switch between the two views should be done by a simple pinch gesture, not by an interface element.
- The default is the current 3 pics per width. The view setting is remembered per folder, and also for subfolders (basically exactly like we do with the grid/list view).

Sounds good?

**Comment by tobiasKaminsky** [comment]
@jancborchardt 
My current approach is
- minimal 3 columns
- increasing by +1/-1 while pinching
- yes, only pinching is needed
- no view setting is being remembered. Do you think this is neccassery? I imagined this feature "just" for quick browsing the files and not having this remembered...?

**Comment by tobiasKaminsky** [comment]
Update: The zoomed grid is being remembered for all grids, but only for the current app session. If one restarts/force closes the app it is set to default column size 3 again.

**Comment by jancborchardt** [comment]
Good stuff. Except I would say that the remembering should be done permanently or not at all. Otherwise it's quite confusing when you restart.

**Comment by tobiasKaminsky** [comment]
@jancborchardt this is now implemented.

**Comment by davigonz** [comment]
@michaelstingl this feature looks interesting and is still used by Google Photos, have a look

![Mar-21-2019 15-13-15](https://user-images.githubusercontent.com/15261363/54758160-f3ba9580-4beb-11e9-9fd7-fa2b820f860a.gif)

And there's a PR (https://github.com/owncloud/android/pull/1466) with the implementation that we could try to merge or at least adapt its code to our current app.

**Comment by michaelstingl** [comment]
> @michaelstingl this feature looks interesting and is still used by Google Photos, have a look

Looks amazing 👍

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
